### PR TITLE
Result processors, resource resolvers,  plus minor fixes

### PR DESCRIPTION
**Pull request by setrofim**

### CODE REVIEW

File: wa/framework/execution.py [hunk 1]
```diff
@@ -95,23 +96,36 @@ def __init__(self, cm, tm, output):
...
-        self.run_state.status = RunStatus.STARTED
+        self.run_state.status = Status.STARTED
+        self.output.status = Status.STARTED
         self.output.write_state()
 
     def end_run(self):
-        self.output.info.end_time = datetime.now()
+        if self.successful_jobs:
+            if self.failed_jobs:
+                status = Satus.PARTIAL
```
Comment (marcbonnici) [inline wa/framework/execution.py]: `Satus.PARTIAL` is misspelt.

File: wa/utils/misc.py [hunk 2]
```diff
@@ -583,3 +583,18 @@ def merge_dicts_simple(base, other):
...
         pass
+
+
+def get_object_name(obj):
+    if hasattr(obj, 'name'):
+        return obj.name
+    elif hasattr(obj, 'im_func'):
+        return '{}.{}'.format(get_object_name(obj.im_class),
+                              obj.im_func.func_name)
+    elif hasattr(obj,'func_name'):
```
Comment (marcbonnici) [inline wa/utils/misc.py]: Missing space after comma.

File: wa/framework/output.py [hunk 3]
```diff
@@ -36,12 +61,17 @@ def reload(self):
...
+
     def add_metric(self, name, value, units=None, lower_is_better=False,
                    classifiers=None):
         self.result.add_metric(name, value, units, lower_is_better, classifiers)
 
     def add_artifact(self, name, path, kind, description=None, classifiers=None):
         if not os.path.exists(path):
+            path = self.get_path(path)
+        if not os.path.exists(path):
             msg = 'Attempting to add non-existing artifact: {}'
```
Comment (setrofim) [inline wa/framework/output.py]: shouldn't it? think about it.

File: wa/processors/csvproc.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    def initialize(self):
+        self.results_so_far = []  # pylint: disable=attribute-defined-outside-init
+        self.artifact_added = False
+
+    def process_job_output(self, output, target_info, run_output):
+        self.results_so_far.append(output)
+        self._write_results(self.results_so_far, run_output)
+        if not self.artifact_added:
+            run_output.add_artifact('run_result_csv', 'results.csv', 'export')
```
Comment (setrofim) [inline wa/processors/csvproc.py]: Run. There is one csv generated for the entire run; but it is updated after every job.

**Comment by setrofim** [comment]
Pushed fixes for other comments.

File: wa/framework/output.py (hunk 3 above)
Comment (marcbonnici) [inline wa/framework/output.py]: Ah right, sorry I should have looked at what get_path did properly.

File: wa/processors/csvproc.py (hunk 4 above)
Comment (marcbonnici) [inline wa/processors/csvproc.py]: Right ok, that makes sense.

**Pull request merged**